### PR TITLE
ONI-84: Additional insuree information panel in claim form.

### PR DESCRIPTION
**Pull request by wzglinieckisoldevelo**
[ONI-84](https://openimis.atlassian.net/browse/ONI-84)

**[REQUIRED PR](https://github.com/openimis/openimis-fe-insuree_js/pull/85)**

Changes:
- Added Panel with additional insuree information (age, fsp, visit duration)
- Added panel with same diagnosis claim
- Added button to previous claims

image

[ONI-84]: https://openimis.atlassian.net/browse/ONI-84?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/ClaimMasterPanelExt.js [hunk 1]
```diff
@@ -40,6 +42,9 @@ class ClaimMasterPanelExt extends Component {
     if (!!claim && !!claim.insuree && !!claim.healthFacility) {
       this.props.fetchLastClaimAt(claim);
     }
+    if (!!claim && !!claim.insuree && !!claim.insuree.chfId && !!claim.icd) {
```
Comment (olewandowski1) [inline src/components/ClaimMasterPanelExt.js:45]: You can shorten this line: `if (!!claim?.insuree?.chfId && !!claim.icd) {...}`

File: src/reducer.js [hunk 2]
```diff
@@ -27,6 +27,10 @@ function reducer(
     fetchedLastClaimAt: false,
     errorLastClaimAt: null,
     lastClaimAt: {},
+    fetchingSameDiagnosisClaim: false,
+    fetchedSameDiagnosisClaim: false,
+    sameDiagnosisClaim: {},
```
Comment (olewandowski1) [inline src/reducer.js:32]: You're using here '{}', but in reducers you're using 'null'. Please, use consistent approach (decide whether you want to stay with '{}' or 'null').

**Review by olewandowski1 — CHANGES_REQUESTED** [review]
@wzglinieckisoldevelo I left 2 comments. Just small changes.

**Review by olewandowski1 — APPROVED** [review]

**Comment by olewandowski1** [comment]
@sniedzielski @wzglinieckisoldevelo I finished this ticket. Could you do a review?

File: src/actions.js [hunk 3]
```diff
@@ -338,6 +338,31 @@ export function clearLastClaimAt() {
   };
 }
 
+export function fetchLastClaimWithSameDiagnosis(icd, chfid) {
+  const claimFilters = [
+    `chfid: "${chfid}"`,
+    `icd: "${icd.code}"`,
+    'status_Gt: 2',
```
Comment (sniedzielski) [inline src/actions.js:345]: what does mean status Gt 2? Can we better describe in code by creating maybe const?

**Review by sniedzielski — COMMENTED** [review]
Good job. I left one comment. Please refer to it @olewandowski1

File: src/actions.js (hunk 3 above)
Comment (olewandowski1) [reply src/actions.js:345]: If I'm not mistaken, this means that I want to receive claims with at least 'entered' status. I will add const to provide more precise naming.

**Comment by olewandowski1** [comment]
@sniedzielski I've addressed your suggestion. Thanks

**Pull request merged**